ui-participant/src/login/RedirectFromOAuth.tsx

```diff
@@ -91,9 +93,31 @@ export const RedirectFromOAuth = () => {
 
             // Enroll in the study if not already enrolled in any other study
             if (defaultEnrollStudy && !loginResult.enrollees.length) {
-              const hubUpdate = await enrollCurrentUserInStudy(defaultEnrollStudy.shortcode,
-                defaultEnrollStudy.name, preEnrollResponseId, refreshLoginState, i18n)
-              navigate('/hub', { replace: true, state: hubUpdate })
+              const hubResponse = await enrollCurrentUserInStudy(
+                defaultEnrollStudy.shortcode, preEnrollResponseId, refreshLoginState)
+
+              const sortedActiveConsentTasks = getSortedActiveTasks(hubResponse.enrollee.participantTasks, 'CONSENT')
+              const nextConsentTask = getNextTask(hubResponse.enrollee, sortedActiveConsentTasks)
```

Comment:
I would pull this into a `getNextConsentTask` function in taskUtils -- this feels like a bit too much logic for the redirect component